engine/src/main/java/org/terasology/world/propagation/SunlightRegenBatchPropagator.java

```diff
@@ -191,90 +178,78 @@ private void reduce(Vector3i position, byte oldValue) {
+            populateMinAdjacent2D(startingRegen, adjStartingRegen, Chunks.SIZE_X, Chunks.SIZE_Z, true);
 
             markForPropagation(adjChunk, depth, startingRegen, adjDepths, adjStartingRegen);
         }
     }
 
     private void markForPropagation(LitChunk toChunk, int[] depth, int[] startingRegen, int[] adjDepths, int[] adjStartingRegen) {
         Vector3i pos = new Vector3i();
-        for (int z = 0; z < ChunkConstants.SIZE_Z; ++z) {
-            for (int x = 0; x < ChunkConstants.SIZE_X; ++x) {
-                int depthIndex = x + ChunkConstants.SIZE_X * z;
+        for (int z = 0; z < Chunks.SIZE_Z; ++z) {
+            for (int x = 0; x < Chunks.SIZE_X; ++x) {
+                int depthIndex = x + Chunks.SIZE_X * z;
                 int start = startingRegen[depthIndex];
                 int adjStart = adjStartingRegen[depthIndex];
+                int initialDepth;
                 if (start - adjStart > 1) {
-                    int initialDepth = Math.max(ChunkConstants.SUNLIGHT_REGEN_THRESHOLD - start, 0);
-                    int finalDepth = depth[depthIndex];
-
-                    int strength = Math.min(start + initialDepth - ChunkConstants.SUNLIGHT_REGEN_THRESHOLD + 1, ChunkConstants.MAX_SUNLIGHT);
-
-                    for (int i = initialDepth; i <= finalDepth; ++i) {
-                        sunlightPropagator.propagateFrom(toChunk.chunkToWorldPosition(x, ChunkConstants.SIZE_Y - i - 1, z),
-                                (byte) (strength));
-                        if (strength < ChunkConstants.MAX_SUNLIGHT) {
-                            strength++;
-                        }
-                    }
+                    initialDepth = Math.max(Chunks.SUNLIGHT_REGEN_THRESHOLD - start, 0);
                 } else {
-                    int initialDepth = Math.max(adjDepths[depthIndex], ChunkConstants.SUNLIGHT_REGEN_THRESHOLD - start);
-                    byte strength = (byte) Math.min(ChunkConstants.MAX_SUNLIGHT, start + initialDepth - ChunkConstants.SUNLIGHT_REGEN_THRESHOLD + 1);
-                    for (int i = initialDepth; i <= depth[depthIndex]; ++i) {
-                        sunlightPropagator.propagateFrom(toChunk.chunkToWorldPosition(x, ChunkConstants.SIZE_Y - i - 1, z), strength);
-                        if (strength < ChunkConstants.MAX_SUNLIGHT) {
-                            strength++;
-                        }
-                        pos.y--;
+                    initialDepth = Math.max(Chunks.SUNLIGHT_REGEN_THRESHOLD - start, adjDepths[depthIndex]);
+                }
+                byte strength = (byte) Math.min(Chunks.MAX_SUNLIGHT, start + initialDepth - Chunks.SUNLIGHT_REGEN_THRESHOLD + 1);
+                for (int i = initialDepth; i <= depth[depthIndex]; ++i) {
+                    sunlightPropagator.propagateFrom(toChunk.chunkToWorldPosition(x, Chunks.SIZE_Y - i - 1, z, pos), strength);
+                    if (strength < Chunks.MAX_SUNLIGHT) {
+                        strength++;
```

Comment:
so you moved the iterator out since its replicated in both conditions

Reply:
yep, I started with doing small changes, bringing arguments in the same order, inlining some parts, and suddenly it was highlighted as duplicate :sweat_smile: